published/2015-07-09-static/static.markdown

```diff
@@ -0,0 +1,168 @@
+---
+tags:
+- swift
+- opensource
+- venmo
+- development
+---
+
+# Static
+
+Today, we open sourced a framework for iOS we've been using a ton internally called [Static](https://github.com/venmo/Static). It's a Swift 2 framework for working with static table views. We use it to power tons of screens in an upcoming project. It's also made prototyping table view-based things super productive.
+
+Static's goal is to separate model data from presentation. `Row`s and `Section`s are your “view models” for your cells. You simply specify a cell class to use and that handles all of the presentation. See the [usage](usage) section below for details.
+
+
+## Building
+
+Static is written in Swift 2 so Xcode 7b3 is required. There aren't any dependencies besides system frameworks.
+
+
+## Installation
+
+[Carthage](https://github.com/carthage/carthage) is the recommended way to install Static. Add the following to your Cartfile:
+
+``` ruby
+github "venmo/Static"
+```
+
+You can also install with [CocoaPods](https://cocoapods.org):
+
+``` ruby
+pod 'Static'
+```
+
+For manual installation, it's recommended to add the project as a subproject to your project or workspace and adding the appropriate framework as a target dependency.
+
+
+## Usage
+
+### Getting Started
+
+To use Static, you need to define [`Row`s](https://github.com/venmo/Static/Static/Row.swift) and [`Section`s](https://github.com/venmo/Static/Static/Section.swift) to describe your data. Here's a simple example:
+
+```swift
+import Static
+
+Section(rows: [
+    Row(text: "Hello")
+])
+```
+
+You can configure `Section`s and `Row`s for anything you want. Here's another example:
+
+```swift
+Section(header: "Money", rows: [
+    Row(text: "Balance", detailText: "$12.00", accessory: .DisclosureIndicator, selection: {
+        // Show statement
+    }),
+    Row(text: "Transfer to Bank…", cellClass: ButtonCell.self, selection: {
+        // Show transfer to bank modal
+    })
+], footer: "Transfers usually arrive within 1-3 business days.")
+```
+
+Since this is Swift, we can instead provide instance methods instead of inline blocks for selections. This makes things really nice. You don't have to switch on index paths in a `tableView:didSelectRowAtIndexPath:` any more!
```

Comment:
`Since this is Swift, we can provide instance methods instead of` (remove one of the insteads)